### ISSUE TITLE
Can the ML-agent do Image segmentation task?

### ISSUE THREAD

**Issue by Xunzhaocunzi** [Unity-Technologies/ml-agents#590]
Hi all, 
I have been looking around for a while, hoping to find an example that do the image segmentation in Unity, but I failed to find one.

I have city model purchased from Unity Asset store, which has a variety of pedestrians, vehicles, and traffic signs. What I want to do is to train the ML-agent to recognize those objects using CNN or similar models. I know how to train the model with Tensorflow/Python alone, but I'd like to train the ML-agent within Unity too.  Thank you very much!

**Comment by mmattar** [comment]
Hi @Xunzhaocunzi - what is the Agent in your scenario and presumably you'd like to just collect the data and pipe it to Python?

For the purpose of transferring your data set from Unity to Python you can use the a visual observation (Camera) to represent the input image and encode the ground-truth pixel labels as a numeric observation.

**Comment by gyanshankar** [comment]
Hi @mmattar - Can you please suggest any example environment for this? I am trying to use visual observation from camera for object detection using tensorflow. The ground truth is going to be a object name and a bounding box of the detected object.

**Comment by xiaomaogy** [comment]
Hi @gyanshankar , we don't have any example environment right now for this. It seems that you are able to transfer your image data from Unity to Python through visual observation? What's blocking yo now?

**Comment by mmattar** [comment]
Hi @Xunzhaocunzi - reconnecting to see if you have any additional follow-ups. As @xiaomaogy mentioned, we do not have example environments for your scenario as the primary task (for now at least) that ML-Agents addresses is reinforcement learning.

**Comment by mmattar** [comment]
Closing this issue for now, but please re-open if you continue facing issues.
